lightning_pose/apps/utils.py

```diff
@@ -0,0 +1,233 @@
+"""Utility functions for streamlit apps."""
+
+import numpy as np
+import pandas as pd
+from pathlib import Path
+import streamlit as st
+import os
+from typing import List, Dict, Tuple
+from collections import defaultdict
+
+pix_error_key = "pixel error"
+conf_error_key = "confidence"
+temp_norm_error_key = "temporal norm"
+pcamv_error_key = "pca multiview"
+pcasv_error_key = "pca singleview"
+
+
+@st.cache(allow_output_mutation=True)
+def update_single_file(curr_file: str, new_file_list: list):
+    if curr_file is None and len(new_file_list) > 0:
+        # pull file from cli args; wrap in Path so that it looks like an UploadedFile object
+        # returned by streamlit's file_uploader
+        ret_file = Path(new_file_list[0])
+    else:
+        ret_file = curr_file
+    return ret_file
+
+
+@st.cache(allow_output_mutation=True)
+def update_file_list(curr_file_list: list, new_file_list: list):
```

Comment:
can remove (see question about commented out lines in `video_diagnostics.py` that use this function)